### ISSUE TITLE
Wav2Vec2: Increasing batch_size doesn't increase nsentences

### ISSUE THREAD

**Issue by weiyengs** [pytorch/fairseq#3284]
Hi,
I tried to pass --batch-size 40 like this: 

`python train.py 'examples/wav2vec/train_data_xlsr_en' --save-dir 'examples/wav2vec/ftune_models_xlsr_en' \
--wer-args '("examples/wav2vec/train_data_xlsr_en/4-gram.bin","examples/wav2vec/train_data_xlsr_en/lexicon.txt",2,-1)' \
--w2v-path examples/wav2vec/ptrain_models/xlsr_53_56k.pt \
--post-process letter --valid-subset valid --no-epoch-checkpoints --best-checkpoint-metric wer --num-workers 0 \
--max-update 13000 --sentence-avg --task audio_pretraining --arch wav2vec_ctc  \
--labels ltr --apply-mask --mask-selection static --mask-other 0 --mask-length 10 --mask-prob 0.1 --layerdrop 0.05 \
--mask-channel-selection static --mask-channel-other 0 --mask-channel-length 64 --mask-channel-prob 0.1 --zero-infinity \
--feature-grad-mult 0.0 --freeze-finetune-updates 10000 --validate-after-updates 10000 --optimizer adam \
--adam-betas '(0.9, 0.98)' --adam-eps 1e-08 --lr 2e-05 --lr-scheduler tri_stage --warmup-steps 1300 --hold-steps 5200 \
--decay-steps 6500 --final-lr-scale 0.05 --final-dropout 0.0 --dropout 0.0 --activation-dropout 0.1 --criterion ctc \
--attention-dropout 0.0 --max-tokens 1280000 --seed 2337 --log-format json --log-interval 10 --ddp-backend no_c10d --normalize \
--fp16 --batch-size 40`

or even --batch-size 100 but my "nsentences" in the updates log still show about 10 at max.
Example output:

`2021-02-26 03:25:20 | INFO | train_inner | {"epoch": 1, "update": 0.886, "loss": "1152.51", "ntokens": "949.1", "nsentences": "10.2", "nll_loss": "12.386", "wps": "1955.7", "ups": "2.06", "wpb": "949.1", "bsz": "10.2", "num_updates": "260", "lr": "4.16e-06", "gnorm": "775.592", "loss_scale": "16", "train_wall": "5", "gb_free": "7.9", "wall": "125"}

2021-02-26 03:25:26 | INFO | train_inner | {"epoch": 1, "update": 0.919, "loss": "1217.66", "ntokens": "1078.9", "nsentences": "10.4", "nll_loss": "11.738", "wps": "2058.2", "ups": "1.91", "wpb": "1078.9", "bsz": "10.4", "num_updates": "270", "lr": "4.31231e-06", "gnorm": "867.413", "loss_scale": "16", "train_wall": "5", "gb_free": "7.8", "wall": "130"}

2021-02-26 03:25:30 | INFO | train_inner | {"epoch": 1, "update": 0.953, "loss": "994.464", "ntokens": "1006.4", "nsentences": "12.3", "nll_loss": "12.154", "wps": "2172.9", "ups": "2.16", "wpb": "1006.4", "bsz": "12.3", "num_updates": "280", "lr": "4.46462e-06", "gnorm": "626.319", "loss_scale": "16", "train_wall": "5", "gb_free": "7.8", "wall": "135"}

2021-02-26 03:25:35 | INFO | train_inner | {"epoch": 1, "update": 0.987, "loss": "1423.31", "ntokens": "993.7", "nsentences": "8.1", "nll_loss": "11.602", "wps": "2135.6", "ups": "2.15", "wpb": "993.7", "bsz": "8.1", "num_updates": "290", "lr": "4.61692e-06", "gnorm": "755.584", "loss_scale": "16", "train_wall": "5", "gb_free": "7.9", "wall": "139"}`

How do I increase the batch size to make full use of my GPU mem?

Thanks!!

**Comment by lematt1991** [comment]
It looks like this task uses [`--max-sample-size`/`--min-sample-size`](https://github.com/pytorch/fairseq/blob/master/fairseq/tasks/audio_pretraining.py#L56-L61) to control the batch size.
